src/libostree/ostree-fetcher-soup.c

```diff
@@ -431,6 +438,24 @@ create_pending_soup_request (OstreeFetcherPendingURI  *pending,
 
   pending->request = soup_session_request_uri (pending->thread_closure->session,
                                                (SoupURI*)(uri ? uri : next_mirror), error);
+
+  /* Add caching headers. */
+  if (SOUP_IS_REQUEST_HTTP (pending->request) && pending->if_none_match != NULL)
+    {
+      glnx_unref_object SoupMessage *msg = soup_request_http_get_message ((SoupRequestHTTP*) pending->request);
+      soup_message_headers_append (msg->request_headers, "If-None-Match", pending->if_none_match);
+    }
+
+  if (SOUP_IS_REQUEST_HTTP (pending->request) && pending->if_modified_since > 0)
+    {
+      glnx_unref_object SoupMessage *msg = soup_request_http_get_message ((SoupRequestHTTP*) pending->request);
+      g_autofree char *mod_date = NULL;
+      g_autoptr(GDateTime) date_time = NULL;
+
+      date_time = g_date_time_new_from_unix_utc (pending->if_modified_since);
+      mod_date = g_date_time_format (date_time, "%a, %d %b %Y %H:%M:%S %Z");
```

Comment:
This kind of stuff I think would look more readable in declare-and-initialize style.  (Not a blocker though)

Reply:
Fixed